### PR TITLE
querySelector/querySelectorAll & executeScript element conversion

### PR DESCRIPTION
**Pull request by scriby**
Add querySelector and querySelectorAll ElementFinderSync methods.

Convert to/from selenium's WebElement class in browserSync.executeScript and browserSync.executeScriptAsync. Note that converting from WebElement to ElementFinderSync in the return value is technically a breaking change, but it seems unlikely to break user code because there wasn't a public API for converting WebElement to ElementFinderSync previously. But it's possible there's some call sites to update in the BB tests.

Increase consistency of executeScript call sites.

@mindywhitsitt the querySelector and querySelectorAll methods are kind of similar in function to the ability protractor-sync used to have for doing instant element selection. I still find that on rare occasion I have a need to do instant element selection. It typically comes up when the test doesn't have a way to know which of several possibilities are going to be on the page and it just needs to try them all and see which one it finds.

One proposal for this is this PR, which provides easy access to the browser's querySelector and querySelectorAll element methods. I think these fit in with some of the other DOM traversal APIs on ElementFinderSync. Given that they're native methods their usage should be well understood.

The fact that these aren't exposed on the main elementSync object will make it a little harder to accidentally use instead of findVisible(s)/findElement(s).

### CODE REVIEW

**Comment by mindywhitsitt** [comment]
Well, I'm not interacting with protractor-sync a lot these days but I believe I get the gist of this. Thank you for the detailed explanation. I think using established API methods whenever possible is the right path. Given that it is technically a breaking change, do you have a recommendation for how we should update the version number? Do you think we should increment the major version for this?

**Comment by scriby** [comment]
Yes, the version should go from 2.x to 3.x. Thanks for taking a look!

**Comment by shermaneric** [comment]
looks good!

**Review by shermaneric — APPROVED** [review]

**Comment by shermaneric** [comment]
If no other comments i'll merge this by end of day and follow up with the version increment bump.

**Comment by shermaneric** [comment]
Sorry nitpicky follow-up on the version information.
Currently we are at `5.2.5` protractor-sync.  

Should we up the Major version to == `6.0.0` ? or 
Minor version to `5.3.0` ...

Based on the conversations above, I'm thinking `6.0.0` but wanted to be sure...
I can also merge this in first and version bump subsequently.

**Comment by scriby** [comment]
I recall a decision where we were trying to stay on the same major version as protractor. I guess that's nice, but following semantic versioning is probably more important. If we don't bump up to 6 then the default npm config (using ^ for the package version) will auto-upgrade projects to this version, which we probably want to avoid due to the breaking change.

We might want to list in the readme what the min version of protractor supported is, or look into using a package.json peerDependency to specify it.

**Comment by shermaneric** [comment]
Thanks @scriby !
I'll merge this here in now and reading up on peerDependencies and will follow up shortly.

**Comment by scriby** [comment]
I don't think I realized that we were specifying protractor as a dependency of this module.

We should definitely get that moved to a devDependency (+ peerDependency) so end users have more control over which version of protractor is in use.

**Comment by shermaneric** [comment]
I don't know if you can see this but my first feeble attempt is [here](https://github.com/blackboard/protractor-sync/pull/70)

I tried locally manipulating further (locally) an interstitial dependency of protractor ( of say 5.3.1) and that installed as expected...

**Comment by scriby** [comment]
I found a bug where stale element re-selection doesn't work with elements selected by querySelector and querySelectorAll. I'll work on this tonight, and it probably makes sense to hold off on the release since there's another change coming in soon, but I'll leave that up to you.

**Pull request merged**
